cpu/fe310/cpu.c

```diff
@@ -19,24 +19,30 @@
 
 #include "cpu.h"
 #include "periph/init.h"
+#include "periph_conf.h"
 
-/**
- * @brief Initialize the CPU, set IRQ priorities, clocks, peripheral
+/*
+ * Configure the memory mapped flash for faster throughput
+ * to minimize interrupt latency on an I-Cache miss and refill
+ * from flash.  Alternatively (and faster) the interrupt
+ * routine could be put in SRAM.  The linker script supports
+ * code in SRAM using the ".hotcode" section.
+ * The flash chip on the HiFive1 is the ISSI 25LP128
+ * http://www.issi.com/WW/pdf/25LP128.pdf
+ * The maximum frequency it can run at is 133MHz in
+ * "Fast Read Dual I/O" mode.
+ * Note the updated data sheet:
+ * https://static.dev.sifive.com/SiFive-FE310-G000-datasheet-v1.0.4.pdf
+ * states "Address and write data using DQ[3] for transmission will not
+ * function properly."  This rules out QPI for the XIP memory mapped flash.
+ * #define MAX_FLASH_FREQ 133000000
+ * On forum SiFive says "safe" operation would be 40MHz.  50MHz seems to work
+ * fine.
  */
-void cpu_init(void)
-{
-    /* Initialize IRQs */
-    irq_init();
+#define MAX_FLASH_FREQ 50000000
 
-    /* Initialize clock */
-    clock_init();
-
-    /* Initialize newlib-nano library stubs */
-    nanostubs_init();
-
-    /* Initialize static peripheral */
-    periph_init();
-}
+/* This should work for any reasonable cpu clock value. */
```

Comment:
What is reasonable? Make range explicit?

Reply:
No idea, this is here only to keep the original code. Note that now it's only used when HFROSC is used.

Reply:
OK, not resolved but posponed

Reply:
I'm not sure to understand, do you mean this won't make it for the release ? (for something that is not used by default and just kept for possible backward compatibility)

Reply:
No, I mean that your PR it not targeting that code, you just moved it around.

Reply:
Nothing to do with the release.

Reply:
Ok, thanks and sorry for the confusion.